### PR TITLE
fix: solve the problem of overwriting file caused by repeated addition

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR
bug
#### What this PR does / why we need it:
fix: solve the problem of overwriting file caused by repeated addition

#### Which issue(s) this PR fixes:
Fixes #

#### Specified Reivewers:
/assgin @your-reviewer

### CODE REVIEW

File: modules/gittar/pkg/gitmodule/repo_edit.go [hunk 1]
```diff
@@ -35,6 +36,13 @@ const (
 	EDIT_PATH_TYPE_BLOB EditPathType = "blob"
 )
 
+var (
+	NotSupportType      = errors.New("not support path type")
+	NotSupportAction    = errors.New("not support action")
+	FileAlreadyExists   = errors.New("存在同名文件")
+	FolderAlreadyExists = errors.New("存在同名文件夹")
+)
+
```
Comment (QvodSoldier) [inline modules/gittar/pkg/gitmodule/repo_edit.go:45]: suggest don't specifically declare global variables here

File: modules/gittar/pkg/gitmodule/repo_edit.go [hunk 2]
```diff
@@ -96,48 +104,64 @@ func (repo *Repository) CreateCommit(request *CreateCommit) (*Commit, error) {
...
 
 	for _, action := range request.Actions {
-		if action.Action == EDIT_ACTION_ADD {
-			if action.PathType == EDIT_PATH_TYPE_TREE {
-				oid, err := rawRepo.CreateBlobFromBuffer([]byte(""))
-				if err != nil {
-					return nil, err
+		switch action.Action {
+		case EDIT_ACTION_ADD, EDIT_ACTION_UPDATE:
+			// 判断文件是否存在，避免文件重复添加
```
Comment (QvodSoldier) [inline modules/gittar/pkg/gitmodule/repo_edit.go:121]: Please use English annotations

File: modules/gittar/pkg/gitmodule/repo_edit.go (hunk 1 above)
Comment (littlejiancc) [reply modules/gittar/pkg/gitmodule/repo_edit.go:45]: refer to https://github.com/uber-go/guide/blob/master/style.md#error-types
image

File: modules/gittar/pkg/gitmodule/repo_edit.go (hunk 2 above)
Comment (littlejiancc) [reply modules/gittar/pkg/gitmodule/repo_edit.go]: ok

**Review by sfwn — APPROVED** [review]

**Pull request merged**